uber/site_sections/mits_applications.py

```diff
@@ -39,6 +41,49 @@ def download_doc(self, session, id):
         cherrypy.response.headers['Content-Disposition'] = 'attachment; filename="{}"'.format(doc.filename)
         return serve_file(doc.filepath, name=doc.filename)
 
+    def check_if_applied(self, session, message='', **params):
+        if cherrypy.request.method == 'POST':
+            subject = c.EVENT_NAME_AND_YEAR + ' MITS Team Confirmation'
+
+            if 'email' not in params:
+                message = "Please enter an email address."
+
+            if not message:
+                last_email = (session.query(Email)
+                              .filter(Email.to.ilike(params['email']))
+                              .filter_by(subject=subject)
+                              .order_by(Email.when.desc()).first())
+                if not last_email or last_email.when < (
+                            localized_now() - timedelta(days=7)):
+                    can_send_email = True
+                else:
+                    can_send_email = False
+
+                mits_teams = session.query(MITSTeam).all()
+
+                match_counter = 0
+                for team in mits_teams:
+                    if params['email'] in team.email:
+                        match_counter += 1
+
+                        if can_send_email:
+                            send_email.delay(
+                                c.MITS_EMAIL,
+                                params['email'],
+                                subject,
+                                render('emails/mits/mits_check.txt',
+                                       {'team': team}, encoding=None),
+                                cc=team.email,
+                                model=team.to_dict('id'))
+
+                if match_counter:
+                    message = 'We found {} team{}.{}'\
+                        .format(match_counter, 's' if match_counter > 1 else '',
+                                ' Please check your email for a link to your application.'
+                                if can_send_email else ' Please check your spam or junk folder.')
```

Comment:
...does this mean if we can't send them e-mail due to them not checking the can-spam box (or whatever it's called now), we tell them to check their spam/junk folder?